Automattic/sensei
# Migrate legacy quiz data

Part of https://github.com/Automattic/sensei/issues/7223

This PR cleans up the code related to a very old quiz data structure (pre version 1.7.4). That code was getting in the way of switching to the new quiz repositories. Also, it was triggering extra queries on each request.

## Proposed Changes

* Add a migration for moving the legacy quiz data to the current data structure.
* Clean up the code and deprecate the use of the legacy quiz data.

## Testing Instructions


1. Install [Sensei LMS 1.7.3](https://github.com/Automattic/sensei/archive/refs/tags/version/1.7.3.zip) on a new site. You may have to turn off the `WP_DEBUG`.
2. Go to Lessons -> All Courses -> Add New and create a course.
3. Go to Questions and add a few questions.
4. Go to Lessons -> Add New. Add a title, choose the course from the "Lesson Course" metabox, and assign the existing questions from the "Existing Questions" tab.
5. Enroll in the course and complete the lesson quiz.
6. Grade the quiz and add some feedback.
7. Check the database. In the `wp_comments` table, there should be a few entries with the `sensei_user_answer` comment type.
8. Disable the Sensei LMS 1.7.3 plugin and install the version from this branch.
9. Once you activate it, the legacy quiz data should be migrated in a few seconds.
10. Verify that the submitted quiz answers are migrated.
11. Verify that the quiz grades and feedback are migrated.

## Deprecated Code


* `Sensei_Utils::sensei_grade_question`
* `Sensei_Utils::sensei_delete_question_grade`
* `Sensei_Utils::sensei_get_user_question_grade`
* `Sensei_Utils::sensei_add_answer_notes`

## Pre-Merge Checklist
